lib/fileSystemS3.js

```diff
@@ -7,21 +7,20 @@ const hostname = require('os').hostname()
 const instanceId = crypto.createHash('sha1').update(hostname + __dirname).digest('hex') // eslint-disable-line no-path-concat
 
 module.exports = ({ logger, accessKeyId, secretAccessKey, bucket, lock = {}, s3Options = {} }) => {
-  if (accessKeyId == null) {
-    throw new Error('The fs store is configured to use aws s3 persistence but the accessKeyId is not set. Use store.persistence.accessKeyId or extensions.fs-store-aws-s3-persistence.accessKeyId to set the proper value.')
-  }
-  if (secretAccessKey == null) {
-    throw new Error('The fs store is configured to use aws s3 persistence but the accousecretAccessKeyntKey is not set. Use store.persistence.secretAccessKey or extensions.fs-store-aws-s3-persistence.secretAccessKey to set the proper value.')
-  }
   if (!bucket) {
     throw new Error('The fs store is configured to use aws s3 persistence but the bucket is not set. Use store.persistence.bucket or extensions.fs-store-aws-s3-persistence.bucket to set the proper value.')
   }
 
-  const s3 = new S3({ accessKeyId: accessKeyId, secretAccessKey: secretAccessKey, ...s3Options })
-  Promise.promisifyAll(s3)
+  let s3
+  if (accessKeyId && secretAccessKey) {
```

Comment:
The current solution is to pass accessKeyId as an empty string '' to get the AWS IAM roles auth. To don't break such devs, here should be instead

accessKeyId != null && secretAccessKey != null